back/spec/acceptance/phases_spec.rb

```diff
@@ -274,6 +274,25 @@
           expect(response_status).to eq 200
         end
       end
+
+      describe "When updating ideation phase with ideas to a poll phase" do
+        before do
+          @project.phases.first.update(
+            participation_method: 'ideation',
+            ideas: create_list(:idea, 2, project: @project)
+          )
+        end
+       
+        let(:id) { @project.phases.first.id }
+        let(:participation_method) { 'poll' }
+
+        example "Existing related ideas_phase remains valid" do
+          expect(@project.ideas[0].ideas_phases.first.valid?).to eq true
```

Comment:
`phase` is already there (which is what the spec is using to find out which phase to update). There should be no need to do e.g. `@project.ideas[0].ideas_phases.first`, you can just replace it by `phase`. But after `do_request` you might need to do `phase.reload` to tell Rails to get the latest values (the new participation method) from the DB.

Reply:
@sebastienhoorens I think `phase` and `ideas_phase` are not the same thing. We are checking the validity of `ideas_phase` which is a through-table entry associated with the idea (linking the idea to a specific `phase_id`). Replacing `@project.ideas[0].ideas_phases.first` with `phase` means the test will pass even when the validation is still applied.

Reply:
I can replace occurrences of `@project.phases.first` with `phase`, however.

Maybe I can also find a way to get a related `idea_phase` through a `phase` too, without needing to reference `@project.ideas`.

Reply:
I have gone with using `let(:ideas_phase) { phase.ideas[0].ideas_phases.first }` to create a reference to specific `IdeasPhase` instance.

Reply:
Oh right, sorry! Introducing an `ideas_phase` variable is indeed a good solution :)